lib/main.dart

```diff
@@ -1,117 +1,84 @@
-  Widget build(BuildContext context) {
-    return MaterialApp(
-      title: 'Flutter Demo',
-      theme: ThemeData(
-        // This is the theme of your application.
-        //
-        // Try running your application with "flutter run". You'll see the
-        // application has a blue toolbar. Then, without quitting the app, try
-        // changing the primarySwatch below to Colors.green and then invoke
-        // "hot reload" (press "r" in the console where you ran "flutter run",
-        // or simply save your changes to "hot reload" in a Flutter IDE).
-        // Notice that the counter didn't reset back to zero; the application
-        // is not restarted.
-        primarySwatch: Colors.blue,
-        // This makes the visual density adapt to the platform that you run
-        // the app on. For desktop platforms, the controls will be smaller and
-        // closer together (more dense) than on mobile platforms.
-        visualDensity: VisualDensity.adaptivePlatformDensity,
-      ),
-      home: MyHomePage(title: 'Flutter Demo Home Page'),
-    );
-  }
+Future<List<Character>> fetchCharacters(http.Client client) async {
+  final response = await client.get('http://swapi.dev/api/people/');
+  return compute(parseCharacter, response.body);
 }
 
-class MyHomePage extends StatefulWidget {
-  MyHomePage({Key key, this.title}) : super(key: key);
-
-  // This widget is the home page of your application. It is stateful, meaning
-  // that it has a State object (defined below) that contains fields that affect
-  // how it looks.
-
-  // This class is the configuration for the state. It holds the values (in this
-  // case the title) provided by the parent (in this case the App widget) and
-  // used by the build method of the State. Fields in a Widget subclass are
-  // always marked "final".
-
-  final String title;
+List<Character> parseCharacter(responseBody) {
+  final parsed = jsonDecode(responseBody) as Map<String, dynamic>;
+  return parsed["results"]
+      .map<Character>((json) => Character.fromJson(json))
+      .toList();
+}
 
+class SWMain extends StatefulWidget {
   @override
-  _MyHomePageState createState() => _MyHomePageState();
+  _SWMainState createState() => _SWMainState();
 }
 
-class _MyHomePageState extends State<MyHomePage> {
-  int _counter = 0;
-
-  void _incrementCounter() {
-    setState(() {
-      // This call to setState tells the Flutter framework that something has
-      // changed in this State, which causes it to rerun the build method below
-      // so that the display can reflect the updated values. If we changed
-      // _counter without calling setState(), then the build method would not be
-      // called again, and so nothing would appear to happen.
-      _counter++;
-    });
-  }
+class _SWMainState extends State<SWMain> {
+  Icon customIcon = Icon(Icons.search);
+  static Text titleText = Text("Star Wars API");
+  Widget customSearchBar = titleText;
+  var search = List<Character>();
```

Comment:
try to use more particular names.
for example searchList.

Reply:
fixed